docs/jotai/api/utils.mdx

```diff
@@ -222,6 +222,35 @@ import deepEqual from 'fast-deep-equal'
 const fooFamily = atomFamily((param) => atom(param), deepEqual)
 ```
 
+### TypeScript
+
+The atom family types will be inferred from initializeAtom.
+It's generally recommended to infer types from initializeAtom.
+You can explicitly declare the parameter, value, and atom's update function types using TypeScript generics.
+
+```ts
+atomFamily<Param, Value, Update>(initializeAtom: (param: Param) => WritableAtom<Value, Update>, areEqual?: (a: Param, b: Param) => boolean)
+atomFamily<Param, Value>(initializeAtom: (param: Param) => Atom<Value>, areEqual?: (a: Param, b: Param) => boolean)
+```
+
+Here's a typical usage with a primitive atom.
+
+```ts
+import type { PrimitiveAtom } from 'jotai'
+
+const myFamily = atomFamily((id: number) => atom(id))
+// the type of atom(id) is PrimitiveAtom<number>
+// PrimitiveAtom<number> is WritableAtom<number, SetStateAction<number>>
+```
+
+If you want to explicitly declare it for a primitive atom, you need to use `SetStateAction`.
```

Comment:
`it` is a bit ambiguous here. Would it help if we explicitly state what we're talking about?

Reply:
```suggestion
If you want to explicitly declare the atomFamily for a primitive atom, you need to use `SetStateAction`.
```
Does this help?

Reply:
I believe so, yes.

Reply:
I believe so, yes.